src/app/FakeLib/TargetHelper.fs

```diff
@@ -322,36 +324,114 @@ let listTargets() =
 // Instead of the target can be used the list dependencies graph parameter.
 let doesTargetMeansListTargets target = target = "--listTargets"  || target = "-lt"
 
+/// Determines a parallel build order for the given set of targets
+let determineBuildOrder (d : TargetTemplate<unit>) =
```

Comment:
This function is quite complex. I think it's a good idea to write some unit tests for it.